### PR TITLE
Adapt to CUDA 11.7 stream memory operations

### LINKED ISSUE #1881: remove usage of CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS_V1
We should check if we can remove the usage of https://github.com/alpaka-group/alpaka/pull/1878#issuecomment-1357759111 for one of our tests with system atomics.

see: https://github.com/alpaka-group/alpaka/pull/1878#issuecomment-1357759111

### PR DESCRIPTION
**Pull request by j-stephan**
Follow-up to #1878. Fixes #1881.

### CODE REVIEW

**Review by bernhardmgruber — DISMISSED** [review]

File: include/alpaka/test/event/EventHostManualTrigger.hpp [hunk 1]
```diff
@@ -437,7 +434,17 @@ namespace alpaka::trait
...
+            using StreamWaitValueFunc = CUresult (*)(CUstream, CUdeviceptr, cuuint32_t, unsigned int);
+#    if(CUDA_VERSION < 11070) || (CUDA_VERSION >= 12000)
+            // NVIDIA introduced a new stream memory ops API with CUDA 11.7 (called v2). The corresponding CUDA
+            // functions were suffixed with `_v2`. With CUDA 12.0 v1 of the API was removed and the `_v2` removed
+            // from the new functions. So CUDA <= 11.6 and CUDA >= 12.0 share the same function signature but
+            // internally do different things.
+            auto streamWaitValue = StreamWaitValueFunc{cuStreamWaitValue32};
+#    else
+            auto streamWaitValue = StreamWaitValueFunc{cuStreamWaitValue32_v2};
+#    endif
```
Comment (bernhardmgruber) [inline include/alpaka/test/event/EventHostManualTrigger.hpp]: Could you please refactor this into a separate function, so the #if logic and comment is not duplicated 3 times? Thx!

**Review by psychocoderHPC — DISMISSED** [review]

File: include/alpaka/test/event/EventHostManualTrigger.hpp (hunk 1 above)
Comment (j-stephan) [reply include/alpaka/test/event/EventHostManualTrigger.hpp]: Done.

**Review by bernhardmgruber — DISMISSED** [review]

File: include/alpaka/test/event/EventHostManualTrigger.hpp [hunk 2]
```diff
@@ -371,21 +371,35 @@ namespace alpaka::test
...
+        // functions were suffixed with `_v2`. With CUDA 12.0 v1 of the API was removed and the `_v2` removed
+        // from the new functions. So CUDA <= 11.6 and CUDA >= 12.0 share the same function signature but
+        // internally do different things.
+#    if(CUDA_VERSION < 11070) || (CUDA_VERSION >= 12000)
+        return cuStreamWaitValue32(stream, addr, value, flags);
+#    else
+        return cuStreamWaitValue32_v2(stream, addr, value, flags);
+#    endif
+    }
+} // namespace alpaka::trait::detail
```
Comment (psychocoderHPC) [inline include/alpaka/test/event/EventHostManualTrigger.hpp]: Someone does not like intentions :rofl:, this code could be added two lines below starting with

```C++
namespace detail
{
...
}
```

**Review by psychocoderHPC — DISMISSED** [review]

File: include/alpaka/test/event/EventHostManualTrigger.hpp [hunk 3]
```diff
@@ -686,13 +701,7 @@ namespace alpaka::trait
             //   on host updates may hang. This includes synchronization between the host and
             //   the device build upon value-based HIP queue synchronization APIs such as
             //   cuStreamWaitValue32() and cuStreamWriteValue32().
-#    if BOOST_COMP_NVCC
-            ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(hipCUResultTohipError(cuStreamWaitValue32(
-                static_cast<CUstream>(queue.getNativeHandle()),
-                reinterpret_cast<CUdeviceptr>(event.m_spEventImpl->m_devMem),
-                0x01010101u,
-                CU_STREAM_WAIT_VALUE_GEQ)));
-#    else
```
Comment (j-stephan) [inline include/alpaka/test/event/EventHostManualTrigger.hpp:695]: We don't support the path alpaka -> HIP -> CUDA anymore so this can be removed entirely.

**Review by bernhardmgruber — APPROVED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**